BrainPortal/lib/boutiques_input_subdir_maker.rb

```diff
@@ -0,0 +1,163 @@
+
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2022
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Some tools expect that a file/directory input is
+# placed in a parent directory.
+#
+# For example we have the following Boutiques command-line:
+#
+#     apptool [PRECOMPUTED_INPUT] [DERIVATIVES_INPUT]
+#
+# with the following input in Boutiques:
+#
+#     {
+#       "description": "A directory that contain the precomputed data process by...",
+#       "id": "precomputed_input",
+#       "name": "Precomputed input",
+#       "optional": false,
+#       "type": "File",
+#       "command-line-flag": "--precomputed",
+#       "value-key": "[PRECOMPUTED_INPUT]"
+#     }
+#     {
+#       "description": "A directory that contain the derivatives data process by...",
+#       "id": "derivatives_input",
+#       "name": "Derivatives input",
+#       "optional": false,
+#       "type": "File",
+#       "command-line-flag": "--derivatives",
+#       "value-key": "[DERIVATIVES_INPUT]"
+#     }
+#
+# and in the `cbrain:integrator_modules` section:
+#
+#     "BoutiquesInputSubdirMaker": {
+#       "precomputed_input": ["precomputed", true],
+#       "derivatives_input": ["derivatives", false]
+#      },
+#
+# In CBRAIN the user will select 2 userfiles for example:
+#
+# For precomputed option `precomputed/sub-n`,
```

Comment:
I don't understand this explanation, in CBRAIN userfiles don't select `precomputed/sub-2` or anything of like that